src/Chapter3.hs

```diff
@@ -970,6 +1086,27 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Weekday
+  = Saturday
+  | Sunday
+  | Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  deriving (Show, Read, Eq, Ord, Bounded, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend Saturday  = True
+isWeekend Sunday    = True
+isWeekend _         = False
+
+nextDay :: Weekday -> Weekday
+nextDay Friday  = Saturday
+nextDay weekday = succ weekday 
+
+daysToParty :: Weekday -> Int
+daysToParty weekday = (fromEnum Friday) - (fromEnum weekday)
```

Comment:
Interesting solution! Because the `Friday` is the last constructor, it's very easy to implement this function :smile:

Reply:
I changed the code not to use my definition of Weekday where Friday is the last day of the week. I change Weekday s.t. Sunday is the last day of the week. I changed the definition of daysToParty not to depend on the last day of the week.